codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesrequest.java

```diff
@@ -12,18 +12,15 @@
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import javax.annotation.Generated;
-import software.amazon.awssdk.core.AmazonWebServiceRequest;
+import software.amazon.awssdk.core.AwsRequestOverrideConfig;
 import software.amazon.awssdk.core.runtime.StandardMemberCopier;
 import software.amazon.awssdk.core.runtime.TypeConverter;
 import software.amazon.awssdk.utils.CollectionUtils;
-import software.amazon.awssdk.utils.builder.CopyableBuilder;
-import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
 
 /**
  */
 @Generated("software.amazon.awssdk:codegen")
-public class AllTypesRequest extends AmazonWebServiceRequest implements
-                                                             ToCopyableBuilder<AllTypesRequest.Builder, AllTypesRequest> {
+public class AllTypesRequest extends JsonProtocolTestsRequest {
```

Comment:
How does this affect `toBuilder()`?

Reply:
What do you mean? Right now, `SdkRequest` defines a `toBuilder` method, which only the leaf type implements.

Reply:
Ah, cool, didn't realize that it was implemented separately from the interface. Could we add the interfaces in the child classes so we have the default methods on the interface as well?

Reply:
Sorry not sure what you meant by your question

Reply:
I mean `public class AllTypesRequest extends JsonProtocolTestsRequest implements ToCopyableBuider<AllTypesRequest.Builder, AllTypesRequest>` and similarly for the builder.

Right now I don't think we can do `AllTypesRequest.builder().apply(...).build()` because the builder doesn't implement `CopyableBuilder`, right?

Reply:
Okay I see. We can change to that, but to be clear it would only be on the leaf type since having the parent classes (like `JsonProtocolTestsRequest` in this example) implement those interfaces too results in a compilation error.

You can actually compile your example since the builders (on the leaf type) expose an `apply` method: https://github.com/dagnir/aws-sdk-java-v2/blob/8a974a91c3d52e58556d84160904e94565e88be9/codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/ModelBuilderSpecs.java#L110 😛